text/0000-bom-in-layer-metadata.md

```diff
@@ -0,0 +1,162 @@
+
+# Motivation
+[motivation]: #motivation
+
+## Why should we do this?
+
+Currently `BOM` is a buildpack level entity and has to be defined in either the `launch.toml` or the `build.toml`. This presents several limitations which may be alleviated the this proposal.
+
+Often times `BOM` describes the artifacts present in the layer. Having this present in the `launch.toml` or `build.toml` which are ephemeral between builds means that the `BOM` has to be generated for each build, even if the layer itself has not changed and is cached. 
+
+This leads to surprising behaviors for users at times when they may be generating a `BOM` as part of the layer generation process and may find that their `BOM` is missing if the `BOM` generation process is not called when the layer is assumed to be cached.
+
+Having the `BOM` be a part of the layer content metadata also means that we can pin-point which layer contributed a BOM entry without having to implement extra logic in downstream buildpacks.
+
+- What is the expected outcome?
+
+The lifecycle is able to generate the final `BOM` by combining the `BOM` entries from top level metadata files like `launch.toml` and `build.toml` and also the ones from `<layer-name>.toml` which handling layer caching correctly i.e. if the layer is cached and re-used in the new build - the `BOM` associated with that layer will also be re-used.
+
+# What it is
+[what-it-is]: #what-it-is
+
+The layer content metadata files would be updated to include - 
+
+```
+[[bom]]
+name = "<dependency name>"
+
+[bom.metadata]
+# arbitrary metadata describing the dependency
+```
+
+For each dependency contributed to the app image or build environment, the buildpack layer:
+
+- SHOULD add a bill-of-materials entry to the bom array describing the dependency, where:
+  - name is REQUIRED.
+  - metadata MAY contain additional data describing the dependency.
+  
+The buildpack SHOULD NOT add bom to a layer content metadata file describing the contents that are not contributed by the same layer. `BOM` entries that describe contribution to the application directory should go in the buildpack-level `BOM` entries in `launch.toml` or `build.toml`.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+The lifecycle would be responsible for the generation of the appropriate metadata files and labels given partial `BOM` entries in the `launch.toml`, `build.toml` and `<layer-name>.toml`. The generation of these artifacts given `BOM` entries in `<layer-name>.toml` would be controlled as follows - 
+
+- If the layer has `types.launch` set to `true` then all the `bom` entries will be contributed to the app image in `io.buildpacks.build.metadata` and `<layers>/config/metadata.toml`.
+- If the layer has `types.build` set to `true` then all the `bom` entries will be written to `report.toml`.
```

Comment:
```suggestion
- If the layer directory exists than all the `bom` entries will be written to `report.toml`.
```
This is what I thought you meant by https://github.com/buildpacks/rfcs/pull/151#discussion_r615155684 and the direction I think we should take things in based on one of the conversation in WG. If the layer directory exists, the layer was present during this build, and therefore the entries go into the build bom, (even if it is an ignored layer).

Reply:
This is covered in L63